src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_agents.clj

```diff
@@ -139,8 +145,11 @@
   [pool-context :- jruby-schemas/PoolContext
    pool-state :- jruby-schemas/PoolState
    refill? :- schema/Bool]
-  (log/info "Draining JRuby pool.")
-  (let [old-instances (collect-all-jrubies pool-state)]
+  (if refill?
+    (log/info "Draining and refilling JRuby pool.")
+    (log/info "Draining JRuby pool."))
+  (let [shutdown-on-error (get-shutdown-on-error-fn pool-context)
+        old-instances (shutdown-on-error #(collect-all-jrubies pool-state))]
```

Comment:
Looking at the current implementation of `collect-all-jrubies`, having a `shutdown-on-error` wrapper here seems okay for now since the only exceptions that would appear to be possible to bubble up here would be for cases where a borrow failed (and a poison pill inserted into the pool).  May need to make the shutdown more surgical in the future, e.g., if/when we implement the lock timeout that we've been talking about for [SERVER-1704](https://tickets.puppetlabs.com/browse/SERVER-1704).